src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -161,7 +161,10 @@ private sealed partial class TagSource
 
                 // Collapse all booleans added to just a max of two ('true' or 'false') representing if we're being
                 // asked for initial tags or not
-                _eventChangeQueue = new AsyncBatchingWorkQueue<bool>(
+                //
+                // PERF: Use AsyncBatchingWorkQueue<bool, VoidResult> instead of AsyncBatchingWorkQueue<bool> because
+                // the latter has an async state machine that rethrows a very common cancellation exception.
```

Comment:
can we not make AsyncBWQ`1 not do that?

Reply:
➡️ No, that change would overlap with the changes from Julien and was eliminated from this pull request. It could be restored at a later time.

Reply:
i would like to see this in a followup.

Reply:
📝 This change was in an earlier version of this pull request, but was removed in response to an earlier code review comment:
https://github.com/dotnet/roslyn/pull/68412#discussion_r1213475945